### PR TITLE
validate_file_using_schema.py accepts a directory

### LINKED ISSUE #1104: validate_file_using_schema.py is incredibly slow
Stunning how slow the CI on gitlab is: e.g. https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters/-/jobs/216303 

This loops over all json files in the model (2140 files) and it takes 30-50 min!!  If would be 1 min using bash, this needs investigation.

### PR DESCRIPTION
**Pull request by orelgueta**
Added the option for the validate_file_using_schema.py application to accept a directory of model parameter json files. This speeds up running on ~2000 files from a few minutes to a few seconds.

Fixes #1104.

### CODE REVIEW

File: simtools/applications/validate_file_using_schema.py [hunk 1]
```diff
@@ -117,6 +130,27 @@ def validate_schema(args_dict, logger):
     logger.info(f"Successful validation of schema file {args_dict['file_name']}")
 
 
+def validate_data_files(args_dict, logger):
+    """Validate data files."""
+    model_parameters_directory = args_dict.get("model_parameters_directory")
+    if model_parameters_directory is not None:
+        mock_args_dict = {}
```
[the remark this answers is not in the document]
Comment (orelgueta) [reply simtools/applications/validate_file_using_schema.py:137]: No problem, changed to `tmp_args_dict`.

**Comment by orelgueta** [comment]
Thanks for the review. Added an integration test now.

**Review by GernotMaier — APPROVED** [review]
All good now thanks

(next step is to change the gitlab CI in the model parameters repository)

**Comment by orelgueta** [comment]
> (next step is to change the gitlab CI in the model parameters repository)

I modified the CI, waiting for this to test it.

**Pull request merged**